mlflow/openai/_openai_autolog.py

```diff
@@ -119,40 +120,30 @@ def patched_call(original, self, *args, **kwargs):
             )
             run_id = run.info.run_id
 
-    log_traces = get_autologging_config(mlflow.openai.FLAVOR_NAME, "log_traces", False)
-    if log_traces:
+    if config.log_traces:
         root_span = mlflow_client.start_trace(
             name=self.__class__.__name__, span_type=_get_span_type(self.__class__), inputs=kwargs
         )
         request_id = root_span.request_id
+        # If a new autolog run is created, associate the trace with the run
+        if run_id is not None:
+            trace_manager = InMemoryTraceManager().get_instance()
+            with trace_manager.get_trace(request_id) as trace:
+                trace.info.request_metadata[TraceMetadataKey.SOURCE_RUN] = run_id
```

Comment:
Will it be helpful if we have a function `set_source_run`?

Reply:
Sounds useful. I think we can add `set_request_metadata` in the trace manager.